src/rgw/rgw_lc.cc

```diff
@@ -1923,13 +1970,94 @@ time_t RGWLC::thread_stop_at()
   return time(nullptr) + interval;
 }
 
+int RGWLC::process_bucket(int index, int max_lock_secs, LCWorker* worker,
+			  const std::string& bucket_entry_marker,
+			  bool once = false)
+{
+  ldpp_dout(this, 5) << "RGWLC::process_bucket(): ENTER: "
+	  << "index: " << index << " worker ix: " << worker->ix
+	  << dendl;
+
+  int ret = 0;
+  rgw::sal::LCSerializer* lock = sal_lc->get_serializer(lc_index_lock_name,
+							obj_names[index],
+							std::string());
+  //string = bucket_name:bucket_id, start_time, int = LC_BUCKET_STATUS
+  rgw::sal::Lifecycle::LCEntry entry;
+  if (max_lock_secs <= 0)
+    return -EAGAIN;
+
+  utime_t time(max_lock_secs, 0);
+retry_lock:
+  ret = lock->try_lock(this, time, null_yield);
+  if (ret == -EBUSY || ret == -EEXIST) {
+    /* already locked by another lc processor */
+    ldpp_dout(this, 0) << "RGWLC::process() failed to acquire lock on "
+		       << obj_names[index] << ", sleep 5, try again" << dendl;
+    sleep(5);
+    goto retry_lock; // XXXX really retry forever?
+  }
+  if (ret < 0)
+    return 0;
+
+  if (! (cct->_conf->rgw_lc_lock_max_time == 9969)) {
```

Comment:
what does this check accomplish? it's not clear where 9969 comes from. there's also a local variable `max_lock_secs` that stores the value of this config option

Reply:
this is debugging magic, try to ignore